src/core/src/op/matmul.cpp

```diff
@@ -2,110 +2,99 @@
     OV_OP_SCOPE(v0_MatMul_visit_attributes);
     visitor.on_attribute("transpose_a", m_transpose_a);
     visitor.on_attribute("transpose_b", m_transpose_b);
     return true;
 }
 
-shared_ptr<Node> op::MatMul::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> MatMul::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v0_MatMul_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<MatMul>(new_args.at(0), new_args.at(1), m_transpose_a, m_transpose_b);
-}
-
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace matmul {
-namespace {
-template <element::Type_t ET>
-bool evaluate(const op::MatMul* op, const HostTensorPtr& arg0, const HostTensorPtr& arg1, const HostTensorPtr& output) {
-    using T = typename element_type_traits<ET>::value_type;
-
-    ov::Shape arg0_shape = arg0->get_shape();
-    ov::Shape arg1_shape = arg1->get_shape();
-
-    std::vector<ov::PartialShape> input_shapes = {arg0_shape, arg1_shape};
-    std::vector<ov::PartialShape> output_shapes = shape_infer(op, input_shapes);
-
-    ov::Shape output_shape = output_shapes[0].to_shape();
-    output->set_element_type(arg0->get_element_type());
-    output->set_shape(output_shape);
-
-    ov::reference::matmul<T>(arg0->get_data_ptr<ET>(),
-                             arg1->get_data_ptr<ET>(),
-                             output->get_data_ptr<ET>(),
-                             arg0_shape,
-                             arg1_shape,
-                             output_shape,
-                             op->get_transpose_a(),
-                             op->get_transpose_b());
-    return true;
-}
-
-bool evaluate_matmul(const op::MatMul* op,
-                     const HostTensorPtr& arg0,
-                     const HostTensorPtr& arg1,
-                     const HostTensorPtr& output) {
-    bool rc = true;
-
-    switch (arg0->get_element_type()) {
-        OPENVINO_TYPE_CASE(evaluate_matmul, i32, op, arg0, arg1, output);
-        OPENVINO_TYPE_CASE(evaluate_matmul, i64, op, arg0, arg1, output);
-        OPENVINO_TYPE_CASE(evaluate_matmul, u32, op, arg0, arg1, output);
-        OPENVINO_TYPE_CASE(evaluate_matmul, u64, op, arg0, arg1, output);
-        OPENVINO_TYPE_CASE(evaluate_matmul, f16, op, arg0, arg1, output);
-        OPENVINO_TYPE_CASE(evaluate_matmul, f32, op, arg0, arg1, output);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
+    return std::make_shared<MatMul>(new_args.at(0), new_args.at(1), m_transpose_a, m_transpose_b);
 }
-}  // namespace
-}  // namespace matmul
 
-bool op::MatMul::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool MatMul::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v0_MatMul_evaluate);
-    return matmul::evaluate_matmul(this, inputs[0], inputs[1], outputs[0]);
+    OPENVINO_ASSERT(outputs.size() == 1);
```

Comment:
do we need to check that there is 2 inputs?

Reply:
Is done in `shape_infer`` function.